csharp/Ice/Filesystem/Client/Program.cs

```diff
@@ -12,35 +12,36 @@
 
 // Recursively list the contents of the root directory.
 Console.WriteLine("Contents of root directory:");
-ListRecursive(rootDir);
+await ListRecursiveAsync(rootDir);
 
 /// <summary>Recursively print the contents of a directory in tree fashion. For files, show the contents of each file.
 /// </summary>
 /// <param name="dir">The directory to list./<param>
 /// <param name="depth">The current nesting level (for indentation).</param>
-void ListRecursive(DirectoryPrx dir, int depth = 0)
+async Task ListRecursiveAsync(DirectoryPrx dir, int depth = 0)
 {
     var indent = new string('\t', ++depth);
 
-    NodePrx?[] contents = dir.List();
+    NodePrx?[] contents = await dir.ListAsync();
 
     foreach (NodePrx? node in contents)
     {
         Debug.Assert(node is not null); // The node proxies returned by list() are never null.
 
+        // TODO: should be an async call
         DirectoryPrx? subdir = DirectoryPrxHelper.checkedCast(node);
         string kind = subdir is not null ? "directory" : "file";
 
-        Console.WriteLine($"{indent}{node.Name()} {kind}:");
+        Console.WriteLine($"{indent}{await node.NameAsync()} {kind}:");
```

Comment:
I would add a name variable here.

Reply:
This demo is really about the power of inheritance. And implementation reuse in C++. So I would not change the model or the `checkedCast[Async]`.